### ISSUE TITLE
Create an invoice with known contact_id without extra API call, without overwriting name

### ISSUE THREAD

**Issue by jrochkind** [waynerobinson/xeroizer#278]
I want to create an Invoice. I know the contact_id of the contact. 

I can create the invoice with the known contact_id like this:

```
  client.Invoice.build(contact: { contact_id: '76ecbf79-6472-4476-aa43-de508258eae0', name: 'some name' }, ...)
```

But I _need_ to supply the name. If I don't, I get a validation error `contact, must be valid`.  If I DO supply the contact name, the name I supply will overwrite the name of the contact on xero, if it was different then the name that was already there. 

I don't want that, I want to leave the contact name on xero alone, not pass it at all, just the contact_id. 

Is this possible? I believe it is with the raw Xero api, is it with Xeroizer?

**Comment by jrochkind** [comment]
For now, it seems like simply removing the `contact` validator on Invoice lets me do what I want to do, as above...

    Xeroizer::Record::Invoice.validators.delete_if {|v| v.attribute == :contact }

Now I can create an invoice with a known contact_id, without needing to know or overwrite the contact name.

**Comment by waynerobinson** [comment]
Fair enough. The API used to care, but it seems it doesn't anymore. I'll
have a look at a potentially more permanent solution.

**Comment by jrochkind** [comment]
You need to supply the contact_id with the invoice creation to the API, but not the contact name. Which makes sense, of course. 

After I've done the `invoice.save`, the full contact is filled out if I look at `invoice.contact`. I'm not sure if another API request happened to fill out that data (and if so, if it happened only lazily after I looked at `invoice.contact`, or happened automatically right after the save), or if the data came back from the invoice post response and xeroizer just used it. 

Anyway, thanks for Xeroizer, it's well designed, which is what at least let me disable the validator so easily to do what I needed!

**Comment by jrochkind** [comment]
Or, I guess if you only supplied the contact name but not an id, it would just create a new contact for you. Perhaps the validator should just check that at least one of id and name are present.

**Comment by timdiggins** [comment]
@jrochkind @waynerobinson this is now fixed by #265
